src/main/java/dinkplugin/notifiers/LootNotifier.java

```diff
@@ -218,6 +219,20 @@ private void handleNotify(Collection<ItemStack> items, String dropper, LootRecor
         }
 
         if (sendMessage) {
+                if (npcId == null && (type == LootRecordType.NPC || type == LootRecordType.PICKPOCKET)) {
+                var player = client.getLocalPlayer();
+                var location = WorldUtils.getLocation(client, player);
+                var comparator = Comparator.<NPC, Boolean>comparing(npc -> npc.getInteracting() == player)
+                    .thenComparing(npc -> npc.isDead() == (type == LootRecordType.NPC))
+                    .thenComparingInt(npc -> -location.distanceTo2D(WorldUtils.getLocation(client, npc)))
+                    .thenComparingInt(NPC::getCombatLevel);
+                npcId = client.getTopLevelWorldView().npcs().stream()
+                    .filter(npc -> dropper.equals(npc.getName()))
+                    .max(comparator)
+                    .map(NPC::getId)
+                    .orElse(null);
```

Comment:
```suggestion
                npcId = client.getTopLevelWorldView().npcs().stream()
                    .filter(npc -> dropper.equals(npc.getName()))
                    .findAny()
                    .map(NPC::getId)
                    .orElse(null);
```

alternatively if we want simpler code at the expense of accuracy in certain edge cases (only matters when there are multiple NPCs loaded with the same name but different IDs)

Reply:
I think the simple solution should be fine, until someone tells us it isn't